FeedStoreChallenge/Core Data/CoreDataFeed.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  CoreDataFeed.swift
+//  FeedStoreChallenge
+//
+//  Created by OLEKSANDR ISAIEV on 13.05.2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+@objc(CoreDataFeed)
+final class CoreDataFeed: NSManagedObject {
+	@NSManaged public var timestamp: Date
+	@NSManaged public var coreDataFeedImages: NSOrderedSet
+}
+
+extension CoreDataFeed {
+	var localFeed: [LocalFeedImage] {
+		return coreDataFeedImages
+			.compactMap { $0 as? CoreDataFeedImage }
+			.map { $0.local }
+	}
+	
+	static func managedFeedSet(from localFeed: [LocalFeedImage], in context: NSManagedObjectContext) -> NSOrderedSet {
+		return NSOrderedSet(array: localFeed.map { feed in
+			let managed = CoreDataFeedImage(context: context)
+			managed.id = feed.id
+			managed.imageDescription = feed.description
+			managed.location = feed.location
+			managed.url = feed.url
+			return managed
+		})
+	}
+}
+
+extension CoreDataFeed {
+	static func fetch(in context: NSManagedObjectContext) throws -> CoreDataFeed? {
+		let request = NSFetchRequest<CoreDataFeed>(entityName: entity().name!)
+		request.returnsObjectsAsFaults = false
+		return try context.fetch(request).first
+	}
+	
+	static func newInstance(in context: NSManagedObjectContext) throws -> CoreDataFeed {
```

Comment:
`newInstance` name is a bit ambiguous as `CoreDataFeed.newInstance(in: context)` has no difference with a plain initialiser `CoreDataFeed(context: context)` from the reader's perspective. 

`uniqueInstance` used during the course adds more context and states that the instance is unique in the context.

Please rename `newInstance` to `uniqueInstance`

```suggestion
	static func uniqueInstance(in context: NSManagedObjectContext) throws -> CoreDataFeed {
```

Reply:
Done